frontends/election-manager/src/components/smartcard_modal/card_details_view.tsx

```diff
@@ -102,7 +97,7 @@ export function CardDetailsView({
       role,
       status: 'InProgress',
     });
-    const result = await card.unprogramUser();
+    const result = await apiClient.unprogramCard();
```

Comment:
I did explore the idea that we discussed in the spec @jonahkagan of updating the methods to take an `authStatus` of a specific type, e.g. `SystemAdministratorLoggedIn`, to ensure that we're in the right state before making calls like this. But I found the resulting code to be complicated and for little gain

Reply:
Sounds good. Maybe we can have some runtime assertions within the auth lib to help enforce those constraints

Reply:
Yeah that makes sense! Will be sure to include those in the `DippedSmartCardAuthWithJavaCard` and `InsertedSmartCardAuthWithJavaCard` implementations